manifests/init.pp

```diff
@@ -1,208 +1,126 @@
-    default => $log_path,
-  }
-
-  $real_log_file_size = $log_file_size ? {
-    'UNSET' => $::teagent::params::log_file_size,
-    default => $log_file_size,
-  }
-
-  $real_proxy_host = $proxy_host ? {
-    'UNSET' => $::teagent::params::proxy_host,
-    default => $proxy_host,
-  }
-
-  $real_proxy_port = $proxy_port ? {
-    'UNSET' => $::teagent::params::proxy_port,
-    default => $proxy_port,
-  }
-
-  $real_proxy_user = $proxy_user ? {
-    'UNSET' => $::teagent::params::proxy_user,
-    default => $proxy_user,
-  }
-
-  $real_proxy_pass = $proxy_pass ? {
-    'UNSET' => $::teagent::params::proxy_pass,
-    default => $proxy_pass,
-  }
-
-  $real_proxy_bypass_list = $proxy_bypass_list ? {
-    'UNSET' => $::teagent::params::proxy_bypass_list,
-    default => $proxy_bypass_list,
-  }
-
-  $real_ip_version = $ip_version ? {
-    'UNSET' => $::teagent::params::ip_version,
-    default => $ip_version,
-  }
-
-  $real_interface = $interface ? {
-    'UNSET' => $::teagent::params::interface,
-    default => $interface,
-  }
-
-  $real_set_repo = $set_repo ? {
-    'UNSET' => $::teagent::params::set_repo,
-    default => $set_repo,
-  }
-
-  # this takes care of the dependencies (repos) if set to true
-  if ($real_set_repo) {
-    include teagent::dependency
-    $teagent_requires = [
-      Class['teagent::dependency'],
-      Class['teagent::repository'],
-    ]
-  }
-  else {
-    $teagent_requires = []
-  }
-
-  package { 'te-agent':
-    ensure  => 'installed',
-    require => $teagent_requires,
-  }
-
-  if $real_browserbot {
-    package { 'te-browserbot':
-      ensure  => 'installed',
-      require => Package['te-agent'],
-    }
-  }
-
-  if $real_agent_utils {
-    package { 'te-agent-utils':
-      ensure  => 'latest',
-      require => Package['te-agent'],
-    }
-  }
-
-  if $real_international_langs {
-    package { 'te-intl-fonts':
-      ensure  => 'latest',
-      require => [
-        Class['teagent::dependency'],
-        Class['teagent::repository'],
-      ],
-    }
-  }
-
-  file { '/var/lib/te-agent/config_teagent.sh':
-    ensure  => 'present',
-    content => template('teagent/config_teagent.sh.erb'),
-    mode    => '0755',
-    require => [ Package['te-agent'], ],
-  }
-
-  exec { '/var/lib/te-agent/config_teagent.sh':
-    subscribe   => File['/var/lib/te-agent/config_teagent.sh'],
-    refreshonly => true,
-  }
+# Copyright © 2017 ThousandEyes, Inc.
+#
+
+class te_agent(
+Optional[String] $proxy_user = undef,
+Optional[String] $proxy_pass = undef,
+Optional[String] $proxy_bypass_list = undef,
+Optional[String] $proxy_location = undef,
+Optional[Enum['BASIC','KERBEROS','NTLM']] $proxy_auth_type = undef,
+Enum['DIRECT','STATIC','PAC'] $proxy_type = 'DIRECT',
+Enum['DEBUG','TRACE'] $log_level = 'DEBUG',
+Integer[0] $log_file_size = 10,
+Integer[0] $num_log_files = 13,
+Integer[0,1] $crash_reports = 1,
+String $log_path = '/var/log',
+String $account_token = 'account_token',
+Boolean $te_agent = true,
```

Comment:
I would change this variable name to something else, like `ensure` or similar. This is because 

`+  $te_agent_package_ensure = $te_agent::te_agent ? {
 +    true  => 'installed',
 +    false => 'purged',
 +  }`

`$te_agent::te_agent` looks kind of weird.

Reply:
What do you think about re-naming the variable to 'agent' instead of 'te_agent'. So the Selector would be $te_agent::agent